File: easy/python/crack/arrays/is_unique.py
# Implement an algorithm to determine if a string has all unique characters
import unittest


def is_unique(string: str):
    i = 0
    length = len(string)
    lower = string.lower()
    while i + 1 < length:
        for item in lower:
            current = item
            if item == current:
                return False
        i += 1

    return True


class TestIsUnique(unittest.TestCase):
    def runTest(self):
        self.assertEqual(is_unique('bingo'), True, 'We expected True')
        self.assertEqual(is_unique('biingo'), False, 'We expected False')
        self.assertEqual(is_unique('Bastardisation'), False, 'We expected False')


unittest.main()

File: crack/arrays/is-unique.js
/*
* @desc Implement an algorithm to determine if a string has all unique characters
* @patterns
*/

const isUnique = (str) => {
    for (let i = 0; i < str.length; i++) {
        for (let j = i + 1; i <= str.length; i++) {
            console.log(str[i] + ' ' + str[j]);
            if (str[i] === str[j]) return false;
        }
    }
    return true;
}

console.log(isUnique('BingoBango')); // False
console.log(isUnique('Bongi')); // True